autogpt_platform/backend/backend/executor/database.py

```diff
@@ -88,10 +88,85 @@ async def lifespan(self, app: "FastAPI"):
             await db.connect()
 
+            # Initialize SQLAlchemy if enabled (for gradual migration from Prisma)
+            if config.enable_sqlalchemy:
+                from sqlalchemy.exc import DatabaseError, OperationalError
+                from sqlalchemy.exc import TimeoutError as SQLAlchemyTimeoutError
+
+                from backend.data import sqlalchemy as sa
+
+                try:
+                    engine = sa.create_engine()
+                    sa.initialize(engine)
+                    app.state.db_engine = engine
+                    logger.info(
+                        f"[{self.service_name}] ✓ SQLAlchemy initialized "
+                        f"(pool_size={config.sqlalchemy_pool_size}, "
+                        f"max_overflow={config.sqlalchemy_max_overflow})"
+                    )
+                except OperationalError as e:
+                    logger.error(
+                        f"[{self.service_name}] Failed to connect to database during SQLAlchemy initialization. "
+                        f"Check database connection settings (host, port, credentials). "
+                        f"Database URL: {config.database_url.split('@')[-1] if '@' in config.database_url else 'N/A'}. "
+                        f"Error: {e}"
+                    )
+                    raise
+                except SQLAlchemyTimeoutError as e:
+                    logger.error(
+                        f"[{self.service_name}] Database connection timeout during SQLAlchemy initialization. "
+                        f"Timeout setting: {config.sqlalchemy_connect_timeout}s. "
+                        f"Check if database is accessible and increase timeout if needed. "
+                        f"Error: {e}"
+                    )
+                    raise
+                except DatabaseError as e:
+                    logger.error(
+                        f"[{self.service_name}] Database error during SQLAlchemy initialization. "
+                        f"Check database permissions and configuration. "
+                        f"Error: {e}"
+                    )
+                    raise
+                except Exception as e:
+                    logger.error(
+                        f"[{self.service_name}] Unexpected error during SQLAlchemy initialization. "
+                        f"Configuration: pool_size={config.sqlalchemy_pool_size}, "
+                        f"max_overflow={config.sqlalchemy_max_overflow}, "
+                        f"pool_timeout={config.sqlalchemy_pool_timeout}s. "
+                        f"Error: {e}",
+                        exc_info=True,
+                    )
+                    raise
+
             logger.info(f"[{self.service_name}] ✅ Ready")
             yield
 
             logger.info(f"[{self.service_name}] ⏳ Disconnecting Database...")
+
+            # Dispose SQLAlchemy if it was enabled
+            if config.enable_sqlalchemy:
```

Comment:
same thing here. But I thought we have created a context/session_manager? why do we still do manual connect/disconnect here.